### PR TITLE
[TE] Alert Page MVP

### PR DESCRIPTION
**Pull request by newsummit**
A more full-featured version of the Alert Page which includes

- Alert metrics (still placeholder)
- Graph placeholder
- Anomaly filters and table sorting/pagination

screen shot 2017-12-07 at 3 24 28 pm

(also made some optimizations in other routes)

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs [hunk 1]
```diff
@@ -1,75 +1,79 @@
...
+          <button {{action "onClickAlertSubscribe"}} class="te-button te-button--main">Subscribe</button>
+        </div>
     </div>
     <ul class="te-search-results__list te-search-results__list--details-block row">
       <div class="col-xs-12 col-sm-5">
         <li class="te-search-results__row">
           {{!-- Todo: Dry this out --}}
-          <div class="te-search-results__option">Metric</div>
-          <div class="te-search-results__value" title={{alertData.metric}}>{{alertData.metric}}</div>
+          <div class="te-search-results__option-solo">Metric</div>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs]: is -solo a modifier or a new block element? 
if it's the former, let's use the correct BEM syntax : `te-search-results__option--solo`

Then the usage will be as follows: (base class + class modifier)
```
class="te-search-results__option te-search-results__option--solo"
```

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs [hunk 2]
```diff
@@ -130,84 +160,188 @@
...
-      {{else}}
-        <div class="te-alert-page-notification">
-          No anomalies detected. Is this an anomaly?
+              </thead>
+              {{else}}
+                No anomalies to show.
+              {{/if}}
+              <tbody>
+                {{#each paginatedFilteredAnomalies as |anomaly|}}
+                  <tr class="te-alert-page-anomalies__table-row te-alert-page-anomalies__table--body">
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs]: fyi, bem need not follow the dom markup. we don't need to start at the page level (te-aler-page.... ) ,

You can just set it to a any block: `anomaly-table__row` or even `table-row__body`

```
Block

Encapsulates a standalone entity that is meaningful on its own. While blocks can be nested and interact with each other, semantically they remain equal; there is no precedence or hierarchy. Holistic entities without DOM representation (such as controllers or models) can be blocks as well.
```
http://getbem.com/naming/

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs [hunk 3]
```diff
@@ -130,84 +160,188 @@
...
+                          options=responseOptions
+                          searchEnabled=false
+                          selected=anomaly.anomalyFeedback
+                          onchange=(action "onChangeAnomalyResponse" anomaly)
+                          as |response|
+                        }}
+                          {{response}}
+                        {{/power-select}}
+
+                        <a target="_blank" class="te-alert-page-anomalies__table-link" href="http://thirdeye.corp.linkedin.com/thirdeye#anomalies?anomaliesSearchMode=id&pageNumber=1&anomalyIds={{anomaly.anomalyId}}">investigate</a>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs]: please remove the `http://thirdeye.corp.linkedin.com` (it's linkedin specific data). 
a relative path is fine (on prod, /app and /thirdeye are on the same level)

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss [hunk 4]
```diff
@@ -122,38 +121,123 @@ body {
...
+  color: app-shade(black, 0.7);
   padding: 24px 0;
 
   &__list {
     list-style: none;
-    // margin-bottom: 24px;
+    font-weight: normal;
+    margin-bottom: 0;
+
+    &--summary-block {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: `--` should be used for modifier such as `--red` `--dark` `--small` `--editable` etc. 
Please use better naming as `--summary-block` `--details-block` indicates another element.

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss [hunk 5]
```diff
@@ -122,38 +121,123 @@ body {
...
+    font-weight: 600;
+    font-size: 14px;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    text-align: right;
+    margin-right: 10px;
+    color: app-shade(black, 4);
+  }
+
+  &__option-solo {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: this seems to be a modifier i.e `--solo`

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss [hunk 6]
```diff
@@ -180,3 +264,42 @@ body {
 .te-link--hide {
   display: none;
 }
+
+.te-button {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: We've added a few styles under` styles/components/button/scss`, can we try to reuse them if possible?

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 7]
```diff
@@ -59,22 +92,150 @@ export default Ember.Controller.extend({
...
-  renderDate: function(range) {
-    // TODO: enable single day range
-    const newDate = moment().subtract(range, 'days').endOf('day').utc().format("DD MM YYY");
-    return `Last ${range} Days (${newDate} to Today)`;
-  },
+  paginationSize: Ember.computed(
+    'pagesNum',
+    'pageSize',
+    function() {
+      const pagesNum = this.get('pagesNum');
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Thanks. Forgot to do this when I added the second property.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 8]
```diff
@@ -83,12 +244,13 @@ export default Ember.Controller.extend({
...
     'totalAnomalies',
     function() {
-      const total = this.get('totalAnomalies');
+      const total = this.get('totalAnomalies') || 0;
       const anomalyStats = [
         {
           title: 'Number of anomalies',
           text: 'Estimated average number of anomalies per month',
           value: total,
+          projected: '5'
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Yes, until we have the endpoint for it.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js [hunk 9]
```diff
@@ -116,108 +142,93 @@ export default Ember.Route.extend({
...
+    }
 
     // Form start/end time ranges based on querystring input with fallback on default '1 month'
-    const durationMap = { 'm':'month', 'd':'day' };
-    const durationMatch = model.duration ? model.duration.match(new RegExp(/^(.\d)+([d|m])$/i)) : [];
-    const queryUnit = durationMatch.length ? durationMap[durationMatch[1].toLowerCase()] : 'month';
-    const durationWindow = calculatedDuration(durationMatch[1] || 1, queryUnit);
-    const baseStart = moment().subtract(durationWindow.time, durationWindow.unit).endOf('day');
-    const baseEnd = moment().subtract(1, 'day').endOf('day');
+    const isCustomDate = qsDuration === 'custom';
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js]: I've pulled this logic out into a function where I describe each default and fall-back step for deriving the final start/end timestamps.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs]: I want it to be a modifier - I've changed the block element and downstream template.

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss (hunk 5 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: yes - fixed

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss (hunk 4 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: yes - fixed

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs (hunk 2 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs]: yes - refactored

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 10]
```diff
@@ -236,15 +529,187 @@ export default Ember.Controller.extend({
...
+     * @param {Object} inputObj - the selection object
+     * @return {undefined}
+     */
+    onChangeAnomalyResponse(anomalyRecord, selectedResponse, inputObj) {
+      const responseObj = _.find(this.get('anomalyResponseObj'), { 'name': selectedResponse });
+      Ember.set(inputObj, 'selected', selectedResponse);
+
+      // Save anomaly feedback
+      this.updateAnomalyFeedback(anomalyRecord.anomalyId, responseObj.value)
+      .then((res) => {
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: done

**Pull request merged**